File: client/csharp/src/ConnectionException.cs
using System;
using System.Runtime.Serialization;

namespace KRPC.Client
{
    /// <summary>
    /// Thrown when a error occurs connecting to a server.
    /// </summary>
    [Serializable]
    public class ConnectionException : Exception
    {
        /// <summary>
        /// Construct an RPCException with no message.
        /// </summary>
        public ConnectionException ()
        {
        }

        /// <summary>
        /// Construct an RPCException with the given message.
        /// </summary>
        public ConnectionException (string message) : base (message)
        {
        }

        /// <summary>
        /// Construct an RPCException with the given message and inner exception.
        /// </summary>
        public ConnectionException (string message, Exception inner) : base (message, inner)
        {
        }

        /// <summary>
        /// Construct an RPCException with the given serialization info and streaming context.
        /// </summary>
        protected ConnectionException (SerializationInfo info, StreamingContext context) : base (info, context)
        {
        }
    }
}

File: client/java/src/krpc/client/ConnectionException.java
package krpc.client;

import java.io.IOException;

/** Thrown when failing to connect to the server. */
public class ConnectionException extends IOException {
  private static final long serialVersionUID = 16427533217488326L;

  /** Creates a new ConnectionException with the given message. */
  public ConnectionException(String message) {
    super(message);
  }

  /** Creates a new ConnectionException with the given message and cause. */
  public ConnectionException(String message, Exception innerException) {
    super(message, innerException);
  }
}
